parsers/Parser.js

```diff
@@ -7,31 +9,97 @@ module.exports = class Parser {
         this.document = document;
     }
 
-    parse() {
-        return [];
+    getRules() {
+        const rules = this.rules;
+
+        return rules && rules.length ? this.flagDuplicates(rules) : [];
```

Comment:
For increased readability, can you wrap the ternary and the condition in parens?

```
return ((rules && rules.length) ? this.flagDuplicates(rules) : []);
```